File: bigbro/tests/symlinkat.py
import tests.helper as th

def passes(out, err):
    return all(
        [th.reads(err, '/tests/symlinkat.test'),
         th.writes(err, '/tmp/new-symlink'),
         th.writes(err, '/tmp/other-link'),
         th.count_writes(err, 2),
         th.count_readdir(err, 0),
     ])

File: bigbro/tests/symlinkat.c
#define _BSD_SOURCE
#define _XOPEN_SOURCE 700

#include <fcntl.h>
#include <unistd.h>

int main(int argc, char **argv) {
  return symlinkat("foobar", AT_FDCWD, "tmp/new-symlink") + symlink("foo", "tmp/other-link");
}
